pywbem/cim_types.py

```diff
@@ -855,38 +860,42 @@ def type_from_name(type_name):
 
 def atomic_to_cim_xml(obj):
     """
-    Convert a value of an atomic scalar CIM data type to a CIM-XML string and
-    return that string.
+    Convert an atomic scalar value to a CIM-XML string and return that string.
 
-    TODO: Verify whether we can change this function to raise a ValueError in
-    case the value is not CIM typed.
+    The returned CIM-XML string is ready for use as the text of a CIM-XML
+    'VALUE' element.
 
     Parameters:
 
-      obj (atomic scalar CIM typed value):
-        The CIM typed value, including `None`. Must be a scalar (not an array).
-        Must be an atomic type (i.e. those listed in :ref:`CIM data types`,
-        except any :ref:`CIM objects`).
+      obj (:term:`CIM data type`, :term:`number`, :class:`py:datetime`):
+        The input value. May be `None`.
+
+        Must not be an array/list/tuple. Must not be a :ref:`CIM object`.
 
     Returns:
 
         A :term:`unicode string` object in CIM-XML value format representing
-        the CIM typed value. For a value of `None`, `None` is returned.
+        the input value. `None`, if the input value is `None`.
+
+    Raises:
+
+        TypeError
     """
-    # pylint: disable=too-many-return-statements
     from .cim_obj import _ensure_unicode  # due to cycles
-    if isinstance(obj, bool):
-        return u"true" if obj else u"false"
-    elif isinstance(obj, CIMDateTime):
+    if obj is None:
+        return obj
+    elif isinstance(obj, six.string_types):
+        return _ensure_unicode(obj)
+    elif isinstance(obj, bool):
+        return u'TRUE' if obj else u'FALSE'
```

Comment:
Question.  Was there any real issue with the lower case true false before.  I agree with the change but just wonder if we fixed any hidden issue

Reply:
DONE
DSP0201 says in 5.3.3.1.4 "Boolean Values":

* These values shall be treated as case-insensitive by CIM-XML consumers. CIM-XML producers should use upper case.

So the change just follows that recommendation, but for consumers that meet the requirement, it should not be a change. However, now that you said it, I have added an according statement to the (already existing) item in the change list (Cleanup section).